web/components/search/query-users.ts

```diff
@@ -0,0 +1,18 @@
+import { User } from 'common/user'
+import { useEffect, useState } from 'react'
+import { searchUsers } from 'web/lib/supabase/users'
+
+// use nonce to make sure only latest result gets used
+let nonce = 0
```

Comment:
It seems a little bit spooky to write it like this, because if you used `useUserSearchResults` in more than one place at the page, everything would be busted. It would make more sense to store this as a `useRef` inside `useUserSearchResults`.